scripts/create_custom_collection.py

```diff
@@ -0,0 +1,86 @@
+import concurrent.futures
+import os
+from pathlib import Path
+
+from scripts.utils import get_genome_ids_with_lineage
+
+"""
+This script creates a custom collection source version for eggNOG CDM data. 
+
+It creates a soft link from the NCBI source data directory to the CDM collection source directory for each genome id 
+with the specified lineage.
+"""
+
+# update the following variables if collection, source version, and target extensions need to be changed
+COLLECTION = 'CDM'
+SOURCE_VER = 'eggNOG'
+TARGET_EXT = ['protein.faa.gz']  # available extensions: 'genomic.fna.gz', 'genomic.gbff.gz', 'protein.faa.gz'
+NUM_THREADS = 128
+
+
+def _create_softlink_between_dirs(
+        new_dir: Path,
+        target_dir: Path,
+        extensions: list[str]
+) -> Path | None:
+    """
+    Establish a symbolic link from new_dir to target_dir, contingent on the presence of files matching the specified extensions.
+
+    return: None if the link is created successfully, otherwise the target_dir.
+    """
+
+    files = os.listdir(target_dir)
+    # Check if all extensions have at least one corresponding file in the target directory
+    found_files = all(any(file.endswith(extension) for file in files) for extension in extensions)
+
+    if not found_files:
+        return target_dir
+
+    if os.path.exists(new_dir) and os.path.islink(new_dir) and os.readlink(new_dir) == str(target_dir):
+        return None
+
+    try:
+        os.symlink(target_dir, new_dir, target_is_directory=True)
+        return None
+    except FileExistsError:
+        raise ValueError(f"{new_dir} already exists and does not link to {target_dir} as expected")
```

Comment:
We have code almost exactly like this in collections... at some point we should think about what code lives where and how we might want to share code

Reply:
yea. This is actually from collection with a slight modification